src/dlls/mscoree/mscoree.cpp

```diff
@@ -792,7 +792,13 @@ BOOL PAL_GetPALDirectory(__out_ecount(cchBuffer) LPWSTR pbuffer,
 
     HRESULT hr = S_OK;
 
-    WCHAR pPath[MAX_LONGPATH];
+    WCHAR * pPath = new (nothrow) WCHAR[MAX_LONGPATH];
+    if (pPath == NULL)
+    {
+        WCHAR pPathStack[MAX_LONGPATH];
```

Comment:
`pPathStack` in your original code is guaranteed to be allocated on stack only by the closing `}` of the scope - `pPath` would be a dangling pointer.